app/models/bit_core/tool.rb

```diff
@@ -29,6 +29,10 @@ def last_position
       content_modules.order(:position).last.try(:position) || 0
     end
 
+    def slug_title
+      slug || title
```

Comment:
If I understand you correctly, it it will initialize content based on the result - but it will never be `NULL`:
`@navigator.initialize_context(context_name)`
i.e.: https://github.com/cbitstech/think_feel_do_engine/blob/5f6b4d202a5691fbc17b866ecd946506cf08d802/app/mailers/think_feel_do_engine/message_notifications.rb#L11-L16
and https://github.com/cbitstech/think_feel_do_engine/blob/5f6b4d202a5691fbc17b866ecd946506cf08d802/app/views/think_feel_do_engine/message_notifications/new_for_participant.html.erb#L49